bigtable/cmd/cbt/cbt.go

```diff
@@ -988,49 +994,91 @@ func doListClusters(ctx context.Context, args ...string) {
 	tw.Flush()
 }
 
-func doLookup(ctx context.Context, args ...string) {
-	if len(args) < 2 {
-		log.Fatalf("usage: cbt lookup <table> <row> [columns=<family:qualifier>...] [cells-per-column=<n>] " +
-			"[app-profile=<app profile id>]")
-	}
+func getDataFilter(
+	parsed map[string]string, filters ...bigtable.Filter,
+) (bigtable.ReadOption, bool, error) {
+	var err error
 
-	parsed, err := parseArgs(args[2:], []string{"columns", "cells-per-column", "app-profile"})
-	if err != nil {
-		log.Fatal(err)
-	}
-	var opts []bigtable.ReadOption
-	var filters []bigtable.Filter
 	if cellsPerColumn := parsed["cells-per-column"]; cellsPerColumn != "" {
 		n, err := strconv.Atoi(cellsPerColumn)
 		if err != nil {
-			log.Fatalf("Bad number of cells per column %q: %v", cellsPerColumn, err)
+			return nil, false, fmt.Errorf(
+				"Bad number of cells per column %q: %v",
+				cellsPerColumn, err)
 		}
 		filters = append(filters, bigtable.LatestNFilter(n))
 	}
+
+	keysOnly := false
+	if keysOnlyS := parsed["keys-only"]; keysOnlyS != "" {
+		keysOnly, err = strconv.ParseBool(keysOnlyS)
+		if err != nil {
+			return nil, false, fmt.Errorf(
+				"Bad value for keys-only: %v", keysOnlyS)
+		}
+		if keysOnly {
+			filters = append(filters, bigtable.StripValueFilter())
+		}
+	}
+
 	if columns := parsed["columns"]; columns != "" {
 		columnFilters, err := parseColumnsFilter(columns)
 		if err != nil {
-			log.Fatal(err)
+			return nil, keysOnly, err
 		}
 		filters = append(filters, columnFilters)
 	}
 
-	if len(filters) > 1 {
-		opts = append(opts, bigtable.RowFilter(bigtable.ChainFilters(filters...)))
-	} else if len(filters) == 1 {
-		opts = append(opts, bigtable.RowFilter(filters[0]))
+	var filter bigtable.Filter = nil
+	var option bigtable.ReadOption
+	if len(filters) > 0 {
+		if len(filters) > 1 {
+			filter = bigtable.ChainFilters(filters...)
+		} else if len(filters) == 1 {
+			filter = filters[0]
+		}
+		option = bigtable.RowFilter(filter)
+	}
+
+	return option, keysOnly, nil
+}
+
+func doLookup(ctx context.Context, args ...string) {
+	if len(args) < 2 {
+		log.Fatalf("usage: cbt lookup <table> <row> [columns=<family:qualifier>...] [cells-per-column=<n>] " +
+			"[app-profile=<app profile id>]")
+	}
+
+	parsed, err := parseArgs(
+		args[2:], []string{
+			"columns", "cells-per-column", "app-profile", "keys-only"},
+	)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	opt, keysOnly, err := getDataFilter(parsed)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	var opts []bigtable.ReadOption
+	if opt != nil {
+		opts = append(opts, opt)
```

Comment:
Should this be `opts = []bigtable.ReadOption{opt}`?

Reply:
No, because `opt` may be nil.

Reply:
Can it though? (line 1063)

Reply:
Oh, I misunderstood your question.  I thought you were referring to line 1062.  Yes, `opts = []bigtable.ReadOption{opt}` is better than using `append`. I'll change that.

Reply:
done